tests/batching_test.py

```diff
@@ -364,7 +364,7 @@ def fun(x, t):
     ans = hessian(lambda x: fun(x, 0.0))(x)
     expected = onp.array([[0., 0., 0., 0., 0.],
                           [0., 0., 0., 0., 0.],
-                          [0., 0.,0.5, 0., 0.],
+                          [0., 0., 2., 0., 0.],
```

Comment:
Can you explain why the new value is more correct than the old one?

Reply:
It should be the second derivative of max(x,0)**2, which is either 2 or 0.
Also, around x=0, the derivative appears to be 1:
```
In [1]: np.maximum(-0.0, 0.0)                                                                                                                       
Out[1]: -0.0

In [2]: np.maximum(0.0, 0.0)                                                                                                                        
Out[2]: 0.0
```

Reply:
Well, x \mapsto max(x, y) is not differentiable at x = y. So we could error, which isn't usually what users expect, or choose some reasonable convention for it. From x < y the directional derivative is 0 and from x > y the directional derivative is 1, and so any value in [0 1] is reasonable by analogy to the subgradient (since max on its own is a convex function). For symmetry with y \mapsto max(x, y) set the result to 0.5.

This numerical test captures the behavior we want, so we shouldn't change its numerical values.